src/gui/EntryPreviewWidget.cpp

```diff
@@ -189,28 +189,31 @@ void EntryPreviewWidget::setPasswordVisible(bool state)
 
 void EntryPreviewWidget::setEntryNotesVisible(bool state)
 {
-    setNotesVisible(m_ui->entryNotesLabel, m_currentEntry->notes(), state);
+    setNotesVisible(m_ui->entryNotesTextEdit, m_currentEntry->notes(), state);
 }
 
 void EntryPreviewWidget::setGroupNotesVisible(bool state)
 {
-    setNotesVisible(m_ui->groupNotesLabel, m_currentGroup->notes(), state);
+    setNotesVisible(m_ui->groupNotesTextEdit, m_currentGroup->notes(), state);
 }
 
-void EntryPreviewWidget::setNotesVisible(QLabel* notesLabel, const QString& notes, bool state)
+void EntryPreviewWidget::setNotesVisible(QPlainTextEdit* notesWidget, const QString& notes, bool state)
 {
+    notesWidget->clear();
+
     if (state) {
         // Add html hyperlinks to notes that start with XXXX://
         QString hyperlinkNotes = notes;
-        notesLabel->setText(hyperlinkNotes.replace(QRegExp("(\\w+:\\/\\/\\S+)"), "<a href=\"\\1\">\\1</a>"));
-        notesLabel->setTextInteractionFlags(Qt::TextBrowserInteraction);
+        hyperlinkNotes.replace("\n", "<br/>");
+        notesWidget->appendHtml(hyperlinkNotes.replace(QRegExp("(\\w+:\\/\\/\\S+)"), "<a href=\"\\1\">\\1</a>"));
```

Comment:
hyperlinkNotes needs proper HTML escaping before doing any replacements and then appending it as HTML to the widget.

Reply:
Good point

Reply:
I also need to fix the styling of this